Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -132,9 +130,38 @@ private void createPaymentAuthRequest(
         boolean isClientTokenAuth = (authorization instanceof ClientToken);
         boolean shouldVault = request.getShouldVault() && isClientTokenAuth;
         sharedPrefsWriter.persistVenmoVaultOption(context, shouldVault);
-        VenmoPaymentAuthRequestParams params =
-                new VenmoPaymentAuthRequestParams(configuration, venmoProfileId, paymentContextId,
-                        braintreeClient.getSessionId(), braintreeClient.getIntegrationType());
+
+        JSONObject braintreeData = new MetadataBuilder()
+                .sessionId(braintreeClient.getSessionId())
+                .integration(braintreeClient.getIntegrationType())
+                .version()
+                .build();
+
+        String applicationName =
+                context.getPackageManager().getApplicationLabel(context.getApplicationInfo())
+                        .toString();
+
+        Uri venmoBaseURL = Uri.parse("https://venmo.com/go/checkout")
+                .buildUpon()
+                .appendQueryParameter("x-success", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/success")
+                .appendQueryParameter("x-error", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/error")
+                .appendQueryParameter("x-cancel", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/cancel")
+                .appendQueryParameter("x-source", applicationName)
+                .appendQueryParameter("braintree_merchant_id", venmoProfileId)
+                .appendQueryParameter("braintree_access_token", configuration.getVenmoAccessToken())
+                .appendQueryParameter("braintree_environment", configuration.getVenmoEnvironment())
+                .appendQueryParameter("resource_id", paymentContextId)
+                .appendQueryParameter("braintree_sdk_data", braintreeData.toString())
```

Comment:
Note that we will want to make sure the metadata changes in both places from PR #945 get make in this PR as well.

Reply:
Updated 👍